src/ztex_md5crypt.c

```diff
@@ -105,6 +105,37 @@ static struct fmt_tests tests[] = {
 };
 
 
+static int cryptmd5_common_valid_salt0(char *ciphertext, struct fmt_main *self)
+{
+	char *pos, *start;
+	char *salt_pos, *salt_end_pos;
+
+	if (!strncmp(ciphertext, md5_salt_prefix, md5_salt_prefix_len))
+		ciphertext += md5_salt_prefix_len;
+	else if (!strncmp(ciphertext, apr1_salt_prefix, apr1_salt_prefix_len))
+		ciphertext += apr1_salt_prefix_len;
+	else if (!strncmp(ciphertext, smd5_salt_prefix, smd5_salt_prefix_len))
+		ciphertext += smd5_salt_prefix_len;
```

Comment:
BTW, we don't have any test vectors for apr1 and smd5 in this format. If we support those hash types in this format, then we should add such test vectors. We should add them _before_ possibly revising the hardware design to support zero-length salts, because adding the test vectors might uncover more issues and shortcomings to fix at the same time.